### PR TITLE
Update React style guide with new rules

### PR DESCRIPTION
**Pull request by goatslacker**
This formalizes the rules that we've been following for class extends React.Component and also updates our ordering with whats in the eslintrc file.

### CODE REVIEW

**Comment by goatslacker** [comment]
Also related: #420

File: react/README.md [hunk 1]
```diff
@@ -219,24 +239,77 @@
...
+    id: PropTypes.number.isRequired,
+    url: PropTypes.string.isRequired,
+    text: PropTypes.string,
+  };
+  
+  const defaultProps = {
+    text: 'Hello World',
+  };
+  
+  class Link extends Component {
```
Comment (spikebrehm) [inline react/README.md]: Alternative: I kind of like doing

```js
export default class Link extends Component {
  ...
}

Link.propTypes = propTypes;
```

thoughts?

**Comment by spikebrehm** [comment]
Nice.

File: react/README.md (hunk 1 above)
Comment (goatslacker) [inline react/README.md]: No opinion on the matter, should we enforce one way or another? It seems like this belongs in the module discussion on the main style guide.

Comment (ljharb) [inline react/README.md]: I like things that encourage people to have a single default export

Comment (iamnirav) [inline react/README.md]: seconded

**Comment by justjake** [comment]
Where are the changes to the .eslint for this? Someone submitted a PR to put `constructor` in the eslint file here: 

I don't want to abandon their PR in favor of this one without the .eslint change from #420 or similar

**Comment by goatslacker** [comment]
No changes to eslintrc in this PR, #420 takes care of adding constructor. The rest of the order is already present in our eslintrc.

**Comment by justjake** [comment]
ok. LGTM

File: react/README.md (hunk 1 above)
Comment (goatslacker) [inline react/README.md]: Looks like we already have that: https://github.com/airbnb/javascript#10.2

**Pull request merged**